src/Network/Response.php

```diff
@@ -717,12 +741,34 @@ public function location($url = null)
 
             return $result;
         }
+        if ($this->_status === 200) {
+            $this->_status = 302;
+        }
         $this->_setHeader('Location', $url);
 
         return null;
     }
 
     /**
+     * Return an instance with an updated location header.
+     *
+     * If the current status code is 200, it will be replaced
+     * with 302.
+     *
+     * @param string $url The location to redirect to.
+     * @return static A new response with the Location header set.
+     */
+    public function withLocation($url)
+    {
+        $new = $this->withHeader('Location', $url);
+        if ($new->_status === 200) {
+            $new->_status = 302;
+        }
```

Comment:
This replicates behavior that used to exist in send() which is not used in the PSR stack.